### PR TITLE
Refactor pg_language out of DatabaseCatalog.

### PR DESCRIPTION
**Pull request by lmwnshn**
Refactor `pg_language` logic away from `DatabaseCatalog` into its own `PgLanguageImpl` object.

Same deal as #1344 which should be merged first.

Just opening it to keep track of merge order.

### CODE REVIEW

File: src/include/catalog/database_catalog.h [hunk 1]
```diff
@@ -36,6 +36,11 @@ class Index;
 
 namespace noisepage::catalog {
 
+namespace postgres {
+class PgLanguageImpl;
+class PgProcImpl;
+}  // namespace postgres
```
Comment (jrolli) [inline src/include/catalog/database_catalog.h:42]: Redundant with including the headers above.

File: src/include/catalog/postgres/pg_language.h [hunk 2]
```diff
@@ -4,34 +4,50 @@
...
+  friend class PgLanguageImpl;
+  friend class PgProcImpl;
+
+  static constexpr table_oid_t LANGUAGE_TABLE_OID = table_oid_t(71);
+  static constexpr index_oid_t LANGUAGE_OID_INDEX_OID = index_oid_t(72);
+  static constexpr index_oid_t LANGUAGE_NAME_INDEX_OID = index_oid_t(73);
+
+  /*
+   * Column names of the form "LAN[name]_COL_OID" are present in the PostgreSQL
+   * catalog specification and columns of the form "ATT_[name]_COL_OID" are
```
Comment (jrolli) [inline src/include/catalog/postgres/pg_language.h:31]:
```suggestion
   * catalog specification and columns of the form "LAN_[name]_COL_OID" are
```

File: src/include/catalog/postgres/pg_language_impl.h [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+  /**
+   * Prepare to create pg_language.
+   * Does NOT create anything until the relevant bootstrap functions are called.
+   *
+   * @param db_oid          The OID of the database that pg_language should be created in.
+   */
+  explicit PgLanguageImpl(db_oid_t db_oid);
+
+  /** Bootstrap the projected row initializers for pg_language. */
+  void BootstrapPRIs();
```
Comment (jrolli) [inline src/include/catalog/postgres/pg_language_impl.h:38]: This should probably be private and called by `PgLanguageImpl::Bootstrap()`.

File: src/include/catalog/postgres/pg_language_impl.h [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+  /**
+   * Bootstrap:
+   *    pg_language
+   *    pg_languages_oid_index
+   *    pg_languages_name_index
```
Comment (jrolli) [inline src/include/catalog/postgres/pg_language_impl.h:44]: Add more details on dependency relationships with other catalog tables (e.g., `pg_proc`?).

**Review by jrolli — CHANGES_REQUESTED** [review]
(Only did review of `pg_language_impl.*` because of current overlap with #1344)

Looks good except for the same high-level comments from #1344:

- [ ] Making `Bootstrap*` as internal details of a single `Bootstrap()` function
- [ ] Friend relationships
- [ ] Resolving where temporary buffers for projected columns live (stack vs heap)

File: src/include/catalog/database_catalog.h (hunk 1 above)
Comment (lmwnshn) [reply src/include/catalog/database_catalog.h]: thanks!

File: src/include/catalog/postgres/pg_language.h (hunk 2 above)
Comment (lmwnshn) [reply src/include/catalog/postgres/pg_language.h]: thanks!

File: src/include/catalog/postgres/pg_language_impl.h (hunk 3 above)
Comment (lmwnshn) [reply src/include/catalog/postgres/pg_language_impl.h:38]: Per previous conversation, no can do right now

File: src/include/catalog/postgres/pg_language_impl.h (hunk 4 above)
Comment (lmwnshn) [reply src/include/catalog/postgres/pg_language_impl.h:44]: wip

**Comment by lmwnshn** [comment]
Will merge with #1358.

**Pull request closed without merging**